packages/lsp-server/src/common/features/diagnostics.ts

```diff
@@ -102,14 +103,23 @@ export class DiagnosticsFeature implements Feature {
 		}
 	}
 
+	private uriForDiagnostics(uri: string): string {
+		if (!this.beanMgr) {
+			return uri;
+		}
+
+		// return as file uri for better integration with beancount diagnostics
+		return 'file://' + URI.parse(uri).fsPath;
+	}
+
 	private async validateAllDocuments(connection: Connection): Promise<void> {
 		await Promise.all(
 			this.documents.all().map(async doc => {
 				await this.validateDocument(doc, connection);
 			}),
 		);
 
-		const documentsInStore = new Set(this.documents.keys());
+		const documentsInStore = new Set(this.documents.keys().map(this.uriForDiagnostics, this));
```

Comment:
https://github.com/fengkx/beancount-lsp/pull/95#issuecomment-3095778232 

filter before map using vscode-uri to parse the string